nlp/optimize/tron.py

```diff
@@ -265,6 +268,11 @@ def projected_newton_step(self, x, g, H, delta, l, u, s, cgtol, itermax):
             step = self.solver.step
             iters += self.solver.niter
 
+            # Exit if the solver took no additional steps
+            if self.solver.niter == 0:
+                exitOptimal = True
```

Comment:
We do. I set info = 4 elsewhere. The exitOptimal = True is just a simplification since the trust region solver treats it as optimal.